### ISSUE TITLE
successCallBack not triggered in iOS

### ISSUE THREAD

**Issue by fattalgazi** [EddyVerbruggen/Toast-PhoneGap-Plugin#76]
Hi there,

```
 window.plugins.toast.showWithOptions(
            {
                message: message,
                duration: duration, // ‘short’, ‘long’
                position: position, // ‘top’, ‘center’, ‘bottom’
                addPixelsY: pikselY,
                styling: 
                {
                    opacity: 1.0, // 0.0 (transparent) to 1.0 (opaque). Default 0.8
                    backgroundColor: bgColor, // make sure you use #RRGGBB. Default #333333
                    // cornerRadius: 16, // minimum is 0 (square). iOS default 20, Android default 100
                    // horizontalPadding: 20, // iOS default 16, Android default 50
                    // verticalPadding: 16 // iOS default 12, Android default 30
                },
                // implement the success callback
                function(result) // jshint ignore:line
                {
                    window.alert('0000');
                    if (result && result.event) 
                    {
                        window.plugins.toast.hide();

                        if(successCallback)
                        {
                            successCallback();
                        }
                    } 
                    // else 
                    // {
                    //     console.log("The toast has been hide, not clicked.");
                    // }
                }
            });
```

window.alert('0000'); will be never triggered.

**Comment by EddyVerbruggen** [comment]
If this is really your code then you're right as it's not correct; the callback handler is part of the options object.

**Comment by fattalgazi** [comment]
sorry for confusion, I didn't understand you, so what is the problem with the code above ?

**Comment by EddyVerbruggen** [comment]
Try it like this:

```js
 window.plugins.toast.showWithOptions(
            {
                message: message,
                duration: duration, // ‘short’, ‘long’
                position: position, // ‘top’, ‘center’, ‘bottom’
                addPixelsY: pikselY,
                styling: 
                {
                    opacity: 1.0, // 0.0 (transparent) to 1.0 (opaque). Default 0.8
                    backgroundColor: bgColor, // make sure you use #RRGGBB. Default #333333
                    // cornerRadius: 16, // minimum is 0 (square). iOS default 20, Android default 100
                    // horizontalPadding: 20, // iOS default 16, Android default 50
                    // verticalPadding: 16 // iOS default 12, Android default 30
                },
            },
                // implement the success callback
                function(result) // jshint ignore:line
                {
                    window.alert('0000');
                    if (result && result.event) 
                    {
                        window.plugins.toast.hide();

                        if(successCallback)
                        {
                            successCallback();
                        }
                    } 
                    // else 
                    // {
                    //     console.log("The toast has been hide, not clicked.");
                    // }
                }
            );
```

**Comment by fattalgazi** [comment]
oh sorry, you are right, thank you man.

**Comment by EddyVerbruggen** [comment]
Does this also fix the opacity issue #75 ?
